kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/conversion/AbstractConverter.java

```diff
@@ -0,0 +1,102 @@
+
+package org.kitodo.mediaserver.core.conversion;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.kitodo.mediaserver.core.api.IConverter;
+import org.kitodo.mediaserver.core.config.ConversionProperties;
+import org.kitodo.mediaserver.core.util.MediaServerUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Abstract converter offering a synchronized method for creating cache file.
+ */
+public abstract class AbstractConverter implements IConverter {
+
+
+    protected boolean saveConvertedFile;
+
+    protected String conversionTargetPath;
+
+    @Autowired
+    protected ConversionProperties conversionProperties;
+
+    @Autowired
+    protected ConversionProperties.Jpeg conversionPropertiesJpeg;
+
+    @Autowired
+    protected MediaServerUtils mediaServerUtils;
+
+    public void setSaveConvertedFile(boolean saveConvertedFile) {
+        this.saveConvertedFile = saveConvertedFile;
+    }
+
+    public void setConversionTargetPath(String conversionTargetPath) {
+        this.conversionTargetPath = conversionTargetPath;
+    }
+
+    public void setMediaServerUtils(MediaServerUtils mediaServerUtils) {
+        this.mediaServerUtils = mediaServerUtils;
+    }
+
+    /**
+     * Checks that all required parameter are present.
+     *
+     * @param master the master file
+     * @param parameter the parameter map
+     * @param requiredParams required parameter
+     * @throws Exception by fatal errors
+     */
+    protected void checkParams(File master, Map<String, String> parameter, String ... requiredParams) {
+
+        if (master == null || !master.exists() || !master.isFile()) {
+            throw new IllegalArgumentException("The master file " + master + " does not exist or is not a file.");
+        }
+
+        mediaServerUtils.checkForRequiredParameter(parameter, requiredParams);
+    }
+
+    /**
+     * Checks if a file exist and creates it, if it doesn't.
+     *
+     * @param file the file
+     * @return true if the file already existed, otherwise false.
+     * @throws IOException if the file could not be created
+     */
+    protected synchronized boolean createCacheFile(File file) throws IOException {
+        if (file == null) {
+            throw new IllegalArgumentException("The file must not be null");
+        }
+        if (file.exists()) {
+            return true;
+        }
+        FileUtils.touch(file);
+        return false;
+    }
+
+    /**
+     * Gets the conversion size from the parameter map. If no parameter is found, it delivers the default size.
+     *
+     * @param parameter the parameter map
+     * @return a size
+     */
+    protected int getConversionSize(Map<String, String> parameter) {
+        try {
+            return Integer.parseInt(parameter.get("size"));
+        } catch (NumberFormatException | NullPointerException e) {
+            return conversionPropertiesJpeg.getDefaultSize();
```

Comment:
Should this be done without any error logging? If the size is not correct because of an error here it is hard to debug this.

Reply:
You're right, I've added a warning